sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/implementation/AzureClientRegistrationRepository.java

```diff
@@ -56,6 +60,14 @@ public boolean isAuthzClient(ClientRegistration client) {
 
     public boolean isAuthzClient(String id) {
         ClientRegistration client = findByRegistrationId(id);
-        return client != null && isAuthzClient(client);
+        return client != null && isAuthzClient(client) && !authorizationProperties.get(id).getOnDemand();
```

Comment:
Maybe we should move ` && !authorizationProperties.get(id).getOnDemand()` to `public boolean isAuthzClient(String id)`.

And enrich `AzureActiveDirectoryConfigurationTest`.